LinkedListNode.swift

```diff
@@ -1,38 +1,56 @@
 
-public class LLNode<T: Equatable & Comparable & CustomStringConvertible> {
+public final class LLNode<T> {
     public var next: LLNode<T>?
     public var value: T
-
-    public init(value: T, next: LLNode<T>? = nil) {
+    
+    public required init(value: T, next: LLNode<T>? = nil) {
         self.value = value
         self.next = next
     }
+}
+
+public protocol Copiable: class {
+    func copy() -> Self
+}
 
-    public func deepCopy() -> LLNode {
-        return LLNode(value: value, next: next?.deepCopy())
```

Comment:
I moved around the copying stuff because this method wasn't actually performing a deep copy over the data structure, since value wasn't guaranteed to be copied if it were a reference type. Therefore, modifying value in one place would still change a copied list.